core/amber/src/main/python/core/runnables/test_network_sender.py

```diff
@@ -0,0 +1,52 @@
+import threading
+from time import sleep
+
+import pytest
+
+from core.models.internal_queue import InternalQueue
+from core.runnables.network_receiver import NetworkReceiver
+from core.runnables.network_sender import NetworkSender
+
+
+class TestNetworkSender:
+    @pytest.fixture
+    def schema_map(self):
+        return dict()
+
+    @pytest.fixture
+    def network_receiver(self, schema_map):
+        network_receiver = NetworkReceiver(InternalQueue(), host="localhost", port=5555, schema_map=schema_map)
+        yield network_receiver
+        network_receiver.stop()
+
+    @pytest.fixture
+    def network_receiver_thread(self, network_receiver):
+        network_receiver_thread = threading.Thread(target=network_receiver.run)
+        yield network_receiver_thread
+
+    @pytest.fixture
+    def network_sender(self, schema_map):
+        network_sender = NetworkSender(InternalQueue(), host="localhost", port=5555, schema_map=schema_map)
+        yield network_sender
+        network_sender.stop()
+
+    @pytest.fixture
+    def network_sender_thread(self, network_sender):
+        network_sender_thread = threading.Thread(target=network_sender.run)
+        yield network_sender_thread
+
+    @pytest.mark.timeout(1)
+    def test_network_sender_can_stop(self, network_receiver, network_receiver_thread, network_sender,
+                                     network_sender_thread):
+        network_receiver_thread.start()
+        network_sender_thread.start()
+        assert network_receiver_thread.is_alive()
+        assert network_sender_thread.is_alive()
+        sleep(0.2)
```

Comment:
I suggest reducing the sleep time to make tests run faster. I'm afraid that little things like this might add up.

Reply:
I reduced the sleep time. In general, python tests are super fast, I do not worry about it for now.